tree/ntupleutil/v7/inc/ROOT/RNTupleExporter.hxx

```diff
@@ -0,0 +1,54 @@
+/// \file ROOT/RNTupleExporter.hxx
+/// \ingroup NTuple ROOT7
+/// \author Giacomo Parolini <giacomo.parolini@cern.ch>
+/// \date 2024-12-10
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RNTupleExporter
+#define ROOT7_RNTupleExporter
+
+#include <filesystem>
+#include <vector>
+
+namespace ROOT::Experimental::Internal {
+
+class RPageSource;
+
+struct RExportPagesOptions {
+   enum RExportPageFlags {
+      kNone = 0x0,
+      kIncludeChecksums = 0x1,   
+      kReportExportedFileNames = 0x2,
+
+      kDefaults = kReportExportedFileNames
+   };
+
+   std::filesystem::path fOutputPath = ".";
```

Comment:
AFAICT the implementation isn't using any of `std::filesystem::path` methods, it just converts to a string. Maybe the options should just take a `std::string` then?

Reply:
For the internal usage std::string would suffice yes, but I think there is value to having a semantically-meaningful type here (since this is user-facing code). For example, a user might want to call some of the path-specific methods on this variable and it's convenient if the variable already is of the "right" type.

Reply:
The type was changed to std::string.